src/tools/PublicAnalyzers/AvaloniaPropertyAnalyzer.cs

```diff
@@ -0,0 +1,463 @@
+        "Unexpected property use: {0} is neither owned by nor attached to {1}",
+        Category,
+        DiagnosticSeverity.Warning,
+        isEnabledByDefault: true,
+        "It is possible to use any AvaloniaProperty with any AvaloniaObject. However, each AvaloniaProperty an object uses on itself should be either owned by that object, or attached to that object.",
+        InappropriateReadWriteTag);
+
+    private static readonly DiagnosticDescriptor SettingOwnStyledPropertyValue = new(
+        "AVP1012",
+        "An AvaloniaObject should not set its own StyledProperty or AttachedProperty values",
+        "Inappropriate assignment: An AvaloniaObject should not set its own StyledProperty or AttachedProperty values",
+        Category,
+        DiagnosticSeverity.Warning,
+        isEnabledByDefault: true,
+        "Setting a StyledProperty or AttachedProperty directly will overwrite values from lower-priority sources, such as styles and templates. This should only be done by control consumers," +
+        "not the control itself. Constructor parameters and assignments within UserControl or TopLevel types are exempt from this diagnostic.",
+        InappropriateReadWriteTag);
+
+    private static readonly DiagnosticDescriptor SuperfluousAddOwnerCall = new(
+        "AVP1013",
+        "AvaloniaProperty owners should not be added superfluously",
+        "Superfluous owner: {0} is already an owner of {1} via {2}",
+        Category,
+        DiagnosticSeverity.Warning,
+        isEnabledByDefault: true,
+        "Ownership of an AvaloniaProperty is inherited along the type hierarchy. There is no need for a derived type to assert ownership over a base type's properties. This diagnostic can be a symptom of an incorrect property owner elsewhere.",
+        InappropriateReadWriteTag);
+
+    private static readonly DiagnosticDescriptor DuplicatePropertyName = new(
+        "AVP1020",
+        "AvaloniaProperty names should be unique within each class",
+        "Name collision: {0} has the same name as {1}",
+        Category,
+        DiagnosticSeverity.Warning,
+        isEnabledByDefault: true,
+        "Querying for an AvaloniaProperty by name requires that each property associated with a type have a unique name.",
+        NameCollisionTag);
+
+    private static readonly DiagnosticDescriptor AmbiguousPropertyName = new(
+        "AVP1021",
+        "There should be an unambiguous relationship between the CLR properties and Avalonia properties of a class",
+        "Name collision: {0} owns multiple Avalonia properties with the name '{1}' {2}",
+        Category,
+        DiagnosticSeverity.Warning,
+        isEnabledByDefault: true,
+        "It is unclear which AvaloniaProperty this CLR property refers to. Ensure that each AvaloniaProperty associated with a type has a unique name. If you need to change behaviour of a base property in your class, call its AddOwner method and provide new metadata.",
```

Comment:
> If you need to change behaviour of a base property in your class, call its AddOwner method and provide new metadata

I think `Property.OverrideMetadata` and `Property.OverrideDefaultValue` are expected here and not AddOwner. Otherwise, it will fail under SuperfluousAddOwnerCall, won't it?